src/server/pfs/server/transaction_defer.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	col "github.com/pachyderm/pachyderm/src/server/pkg/collection"
+	txnenv "github.com/pachyderm/pachyderm/src/server/pkg/transactionenv"
+)
+
+// TransactionDefer is an object that is used to defer certain cleanup tasks
+// until the end of a transaction.  The transactionenv package provides the
+// interface for this and will call the Run function at the end of a
+// transaction.
+type TransactionDefer struct {
+	d   *driver
+	stm col.STM
+
+	// Branches to propagate when the transaction completes
+	propagateBranches []*pfs.Branch
+}
```

Comment:
The name defer seems a little bit weird here because as far as I could tell this doesn't actually get deferred anywhere. I'm not sure I have a better name though.

Reply:
Renamed to `Propagater` so we can defer the bikeshedding until this type actually needs to do anything else but propagate commits.